lib/teleterm/daemon/daemon.go

```diff
@@ -789,6 +789,18 @@ func (s *Service) AssumeRole(ctx context.Context, req *api.AssumeRoleRequest) er
 		return trace.Wrap(err)
 	}
 
+	for _, gw := range s.gateways {
```

Comment:
This needs to acquire a mutex so that gateways are not modified while we're iterating over them.

Reply:
Also, we should explain here why we're doing this.